papermill/s3.py

```diff
@@ -406,36 +493,37 @@ def cat(self,
 
             if size != bytes_read:
                 if err:
-                    raise
+                    raise Exception
```

Comment:
Ditto on raising e from earlier. This might warrant more of a refactor to reduce the function complexity to achieve that here though